### PR TITLE
spin VM in AWS

### LINKED ISSUE #126: spin VM in aws and add ubuntu machine to Virtualbox

### PR DESCRIPTION
**Pull request by BarakMerdler**
fix #126 

Things missing:

1. create ubuntu VM to run on VirtualBox
2. crate bootstrap.sh for ubuntu VM

Done!

### CODE REVIEW

**Comment by BarakMerdler** [comment]
Ready for review @lioramilbaum

**Comment by lmilbaum** [comment]
When PR is WIP, use the 'WIP:' prefix in the PR subject. Do not use it in the comment.

**Review by lmilbaum — CHANGES_REQUESTED** [review]
please use the Multi-Machine mechanism. The way it is implemented causes lots of challenges when running various commands, like: status

**Comment by BarakMerdler** [comment]
I used the WIP in the PR subject, but now it's ready for review.

I editet the first comment.

**Comment by BarakMerdler** [comment]
@lioramilbaum ready for 2nd review

File: README.md [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+
+- Go to your AWS console -> EC2 -> Key pairs -> Create key pair
+- Name: tiger + File format: pem
+- Move the file from Downloads to ~/.ssh
+- Windows users please address the additional steps before you countine
+- run 'vagrant up stage'
+
+# Additional steps for Windows users
+
+- Move to the tiger folder
```
Comment (lmilbaum) [inline README.md]: Not really required. The plugin installation is NOT per project. It is per user who is running the plugin installation.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,43 @@
...
+3. Run 'vagrant up [fedora OR ubuntu OR stage]'
+4. Run 'vagrant ssh [fedora OR ubuntu OR stage]'
+5. In the end of work, logout from the machine
+6. Run 'vagrant destroy -f [fedora OR ubuntu OR stage]'
+
+#Choosing OS
+
+- For fedora OS run 'vagrant up fedora'
+- For ubuntu OS run 'vagrant up ubuntu'
+- When you ssh into machine specify the name of the machine [fedora OR ubuntu OR stage]'
```
Comment (lmilbaum) [inline README.md]: 'vagrant ssh [ubuntu|fedora|stage]'

File: Vagrantfile [hunk 3]
```diff
@@ -1,8 +1,44 @@
...
+end
+
+  config.vm.define "stage" do |stage|
+    stage.vagrant.plugins = "libxml2-dev"
+    stage.vagrant.plugins = "vagrant-aws"
+    stage.vm.box = "dummy"
+    stage.vm.box_url = "https://github.com/mitchellh/vagrant-aws/raw/master/dummy.box"
+    stage.vm.provider :aws do |aws, override|
+    stage.vm.synced_folder ".", "/vagrant", 
+      disabled: false
```
Comment (lmilbaum) [inline Vagrantfile]: Is it required? Isn't it enabled by default?

File: scripts/bootstrap.sh [hunk 4]
```diff
@@ -1,20 +1,59 @@
 #!/usr/bin/env bash
```
Comment (lmilbaum) [inline scripts/bootstrap.sh:1]: There is a lot of code repetition, both here and in the Vagrantfile.  We can work on it together if it can be of help.

**Review by lmilbaum — CHANGES_REQUESTED** [review]
I didn't find the issue # in the first PR comment.

File: scripts/bootstrap.sh (hunk 4 above)
Comment (BarakMerdler) [reply scripts/bootstrap.sh:1]: I will be happy to sit on it with you

File: README.md (hunk 1 above)
Comment (BarakMerdler) [reply README.md]: ack

File: README.md (hunk 2 above)
Comment (BarakMerdler) [reply README.md]: ack

File: Vagrantfile (hunk 3 above)
Comment (BarakMerdler) [reply Vagrantfile]: For running the bootstrap.sh I moved this line to the start of the file and now is working, when I deleted the line it didn't work

**Comment by BarakMerdler** [comment]
ready for 3rd review @lioramilbaum 
travis-ci not passing because of 'app_functions.py'

**Comment by lmilbaum** [comment]
@BarakMerdler The CI should be fixed before I can continue and approve new PRs.

**Comment by BarakMerdler** [comment]
used git commit --allow-empty -m "Trigger" to trigger travis
now passed all test
@lioramilbaum

**Review by lmilbaum — APPROVED** [review]
LGTM

**Pull request merged**